### PR TITLE
MINOR: Server-Commons cleanup

### PR DESCRIPTION
**Pull request by jlprat**
Fixes Javadoc and minor issues in the Java files of Server-Commons modules.

Javadoc is now formatted as intended by the author of the doc itself.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server-common/src/main/java/org/apache/kafka/queue/KafkaEventQueue.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ private void remove(EventContext eventContext) {
             }
         }
 
-        private void handleEvents() throws InterruptedException {
```
Comment (jlprat) [inline server-common/src/main/java/org/apache/kafka/queue/KafkaEventQueue.java:204]: No call in the method throws `InterruptedException`

File: server-common/src/main/java/org/apache/kafka/timeline/BaseHashTable.java [hunk 2]
```diff
@@ -180,7 +180,7 @@ final T baseRemove(Object key) {
     /**
      * Expand the hash table to a new size.  Existing elements will be copied to new slots.
      */
-    final private void rehash(int newSize) {
```
Comment (jlprat) [inline server-common/src/main/java/org/apache/kafka/timeline/BaseHashTable.java:183]: This is not needed

File: server-common/src/main/java/org/apache/kafka/timeline/BaseHashTable.java [hunk 3]
```diff
@@ -224,15 +224,15 @@ static int findSlot(Object object, int numElements) {
      */
     static <T> void unpackSlot(List<T> out, Object[] elements, int slot) {
         Object value = elements[slot];
-        if (value == null) {
-            return;
```
Comment (jlprat) [inline server-common/src/main/java/org/apache/kafka/timeline/BaseHashTable.java:228]: This is done to avoid a return on a void method

**Comment by jlprat** [comment]
Hi @mimaison you were doing some similar changes recently, if you have some time, I'd highly appreciate a review

File: server-common/src/main/java/org/apache/kafka/server/util/timer/Timer.java [hunk 4]
```diff
@@ -27,7 +27,7 @@ public interface Timer extends AutoCloseable {
     /**
      * Advance the internal clock, executing any tasks whose expiration has been
      * reached within the duration of the passed timeout.
-     * @param timeoutMs
+     * @param timeoutMs the tme to advance in milliseconds
```
Comment (mimaison) [inline server-common/src/main/java/org/apache/kafka/server/util/timer/Timer.java]: `tme` -> `time`

File: server-common/src/main/java/org/apache/kafka/server/util/ThroughputThrottler.java [hunk 5]
```diff
@@ -64,7 +64,7 @@ public ThroughputThrottler(long targetThroughput, long startMs) {
      * @param amountSoFar bytes produced so far if you want to throttle data throughput, or
      *                    messages produced so far if you want to throttle message throughput.
      * @param sendStartMs timestamp of the most recently sent message
-     * @return
+     * @return <code>true</code> if throttling happened
```
Comment (mimaison) [inline server-common/src/main/java/org/apache/kafka/server/util/ThroughputThrottler.java]: Should it be `<code>true</code> if throttling should happen`?

File: server-common/src/main/java/org/apache/kafka/server/util/json/JsonValue.java [hunk 6]
```diff
@@ -26,18 +26,16 @@
 /**
  * A simple wrapper over Jackson's JsonNode that enables type safe parsing via the `DecodeJson` type
  * class.
- *
+ * <br>
  * Typical usage would be something like:
- *
- * {{{
+ * <pre><code>
```
Comment (mimaison) [inline server-common/src/main/java/org/apache/kafka/server/util/json/JsonValue.java:31]: It looks like we forgot to update the code example when we converted this class to Java, the snippet below is still in Scala.

**Review by mimaison — COMMENTED** [review]
Thanks for the PR. I left a couple of comments

File: server-common/src/main/java/org/apache/kafka/server/util/json/JsonValue.java (hunk 6 above)
Comment (jlprat) [reply server-common/src/main/java/org/apache/kafka/server/util/json/JsonValue.java:31]: Oh good catch! I'll update this and the other feedback

Comment (jlprat) [reply server-common/src/main/java/org/apache/kafka/server/util/json/JsonValue.java:31]: Done! Thanks for the feedback @mimaison !

**Review by mimaison — APPROVED** [review]
Thanks, LGTM

**Comment by jlprat** [comment]
I mistakenly added 2 imports in the last commit (added when writing the example code)

**Comment by jlprat** [comment]
All tests seem to be unrelated to the change

**Pull request merged**